flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/NetworkBufferPool.java

```diff
@@ -162,6 +162,10 @@ public void recycle(MemorySegment segment) {
 
 	@Override
 	public List<MemorySegment> requestMemorySegments() throws IOException {
+		if (numberOfSegmentsToRequest <= 0) {
```

Comment:
`numberOfSegmentsToRequest`  should never be negative because we already check this argument in constructor.
`if (numberOfSegmentsToRequest == 0)` instead?